Street Trees/STTreeMapViewController.swift

```diff
@@ -235,11 +242,22 @@ class STTreeMapViewController: UIViewController, MKMapViewDelegate, CLLocationMa
         }
         
         self.backgroundQueue.addOperationWithBlock { [weak self] in
-            STTKDownloadManager.fetch(cityGeoPoints: { (response: [AnyObject]) in
-                if let polygons = response as? [MKPolygon] {
-                    self?.mapView.addOverlays(polygons)
+            STPKCoreData.sharedInstance.fetchCityBounds({ (cityBounds, error) in
+                guard let bounds = cityBounds else { return }
+                
+                defer {
+                    UIApplication.sharedApplication().networkActivityIndicatorVisible = false
+                }
+                
+                do {
+                    let shapes = try bounds.shapes()
+                    NSOperationQueue.mainQueue().addOperationWithBlock({ [weak self] in
+                        self?.mapView.addOverlays(shapes)
+                    })
+
+                } catch {
+                    
```

Comment:
Were you planning to handle an exception later or is this here so that the app doesn't crash if the try throws?